src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -137,8 +137,9 @@ trait SubmissionComponent {
       def saveSubmissionWorkflowFailures(workflowFailures: Seq[WorkflowFailure]) = {
         DBIO.seq(workflowFailures.map { case (wff) =>
           for {
-            entityId <- loadSubmissionEntityId(workspaceContext.workspaceId, submission.submissionEntity)
-            failureId <- (workflowFailureQuery returning workflowFailureQuery.map(_.id)) += marshalWorkflowFailure(entityId, UUID.fromString(submission.submissionId))
+            wfEntityId <- uniqueResult[EntityRecord](entityQuery.findEntityByName(workspaceContext.workspaceId, wff.entityType, wff.entityName))
```

Comment:
Nit: can you pull this for comprehension out into something like `workflowQuery.saveFailures` so it looks like the other block immediately above?